docs/specs/app-initiated-sso-to-web.md

```diff
@@ -0,0 +1,182 @@
+
+### Token Exchange Request
+
+When a mobile app wants to perform App-initiated SSO to Web, it makes a request to the token endpoint to obtain a device browser session token with the following parameters:
+
+- `grant_type`: Must be `urn:ietf:params:oauth:grant-type:token-exchange`
+- `subject_token`: The id_token obtained by the mobile app
+- `subject_token_type`: Must be `urn:ietf:params:oauth:token-type:id_token`
+- `actor_token`: The device_secret obtained by the mobile app
+- `actor_token_type`: Must be `urn:x-oath:params:oauth:token-type:device-secret`
+- `requested_token_type`: Required. The value must be `urn:authgear:params:oauth:token-type:device-browser-session-token`.
+
+Here is an example:
+
+```
+POST /token HTTP/1.1
+Host: auth.example.com
+...
+grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange
+&subject_token=<id_token>
+&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aid-token
+&actor_token=95twdf3w4y6wvftw35634t
+&actor_token_type=urn%3Ax-oath%3Aparams%3Aoauth%3Atoken-type%3Adevice-secret
+&requested_token_type=urn%3Aauthgear%3Aparams%3Aoauth%3Atoken-type%3Adevice-browser-session-token
+```
+
+### Token Exchange Response
+
+The Token Exchange response for this profile has the following characteristics:
+
+- `access_token`: Required. The device browser session token which can then be used to exchange for a browser cookie. It is named as `access_token` for historical reasons as mentioned in [RFC8963](https://datatracker.ietf.org/doc/html/rfc8693#section-2.2.1).
+- `issued_token_type`: Required. The value must be `urn:authgear:params:oauth:token-type:device-browser-session-token`.
+- `token_type`: Required. The value is `Bearer` if DPoP header is not sent in the request, else the value is `DPoP`. See the [Binding Tokens To Device](#binding-tokens-to-device) for details on DPoP.
+- `device_secret`: Required. A new device secret must be returned. The mobile app should update its device secret with the new secret.
+- `expires_in`: Required. When the device browser session token expires.
+
+Here is an example:
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "device_secret":"casdfgarfgasdfg",
+  "x_device_browser_session_token": "zxczxczxc",
+  "expires_in":300
+}
+```
+
+## Exchanging for browser cookie of a session with the authorization endpoint
+
+A web browser can open the authorization endpoint endpoint to exchange for a valid browser session cookie with the following parameters:
+
+- `client_id`: Required. The client_id of the obtained session.
+- `id_token_hint`: Required. The id_token obtained by the mobile app.
+- `x_device_browser_session_token`: Required. The device browser session token obtained from token exchange response.
+- `scope`: Optional. Scope of the session of the browser session cookie. If not provided, it will be same as the scope of the session which generates this browser session cookie (i.e. The offline grant of the mobile app).
```

Comment:
No scope here, because it was specified in previous token exchange step already.